sys/net/network_layer/fib/fib.c

```diff
@@ -363,10 +363,12 @@ static int fib_signal_rp(fib_table_t *table, uint16_t type, uint8_t *dat,
                 }
             }
             else {
-                fib_sr_t *temp_sr = (fib_sr_t *)dat;
-                size_t dat_size_in_bits = temp_sr->sr_dest->address->address_size << 3;
+                fib_sr_entry_t sr_dest;
+                assert(dat_size >= sizeof(fib_sr_t));
+                memcpy(&sr_dest, ((fib_sr_t *)(uintptr_t)dat)->sr_dest, sizeof(fib_entry_t));
+                size_t dat_size_in_bits = sr_dest.address->address_size << 3;
```

Comment:
I mean like 
```suggestion
                fib_sr_t *temp_sr = (fib_sr_t *)(uintptr_t)dat;
                fib_sr_entry_t *sr_dest;
                memcpy(&sr_dest, &temp_sr->sr_dest, sizeof(sr_dest));
                size_t dat_size_in_bits = sr_dest->address->address_size << 3;
```

Reply:
But I guess if we trust the byte array to contain valid pointers within the local memory, we can just as well trust it to be correctly aligned.